src/scripts/setup.sh

```diff
@@ -12,6 +12,25 @@ Install() {
     if [ "$UID" = "0" ]; then export SUDO=''; else export SUDO='sudo'; fi
     $SUDO find /etc/datadog-agent/conf.d/ -iname "*.yaml.default" -delete
     $SUDO service datadog-agent start
+
+    set +e
+    attempts=0
+    $SUDO datadog-agent health
+    exit_code=$?
+    until [[ $attempts -ge 10 ||  $exit_code -eq 0 ]]; do
+        attempts=$((attempts+1))
+        sleep_time=$(( attempts*10 < 60 ? attempts*10 : 60 ))
+        echo "Waiting for agent to start up sleeping for ${sleep_time} seconds"
+        sleep $sleep_time
+
+        $SUDO datadog-agent health
+        exit_code=$?
+    done
+
+    if [[ $exit_code -ne 0 ]]; then
+        echo "Could not start the agent"
+        exit 1
```

Comment:
```suggestion
        exit $exit_code
```
Maybe better if `datadog-agent health` provides any detailed codes?